examples/opt/schedule.py

```diff
@@ -197,7 +197,10 @@ def pattern(x: torch.Tensor) -> torch.Tensor:
             sub_sch["module.FusedQKV_0.fused_linear"].shard("bias", axis=0)
             sub_sch["module.FusedQKV_0.fused_linear"].sync(mode="backward")
             sub_sch["module.out_proj"].shard("weight", axis=1)
-            sub_sch["module.out_proj"].sync(mode="forward")
+            sub_sch["module.out_proj"].sync(
+                mode="forward_defer_gather",
+                gather_at=(sub_sch["module.resid_dropout"], 1),
```

Comment:
Can we pass in two kwargs instead of a tuple? The number without a name looks weird.

Reply:
This is refactored per offline discussion with @szhengac.